consensus/XDPoS/engines/engine_v1/engine.go

```diff
@@ -956,7 +958,10 @@ func (x *XDPoS_v1) Seal(chain consensus.ChainReader, block *types.Block, stop <-
 // that a new block should have based on the previous blocks in the chain and the
 // current signer.
 func (x *XDPoS_v1) CalcDifficulty(chain consensus.ChainReader, time uint64, parent *types.Header) *big.Int {
-	return x.calcDifficulty(chain, parent, x.signer)
+	x.lock.Lock()
+	signer := x.signer
+	x.lock.Unlock()
```

Comment:
Can we use x.lock.RLock() and x.lock.RUnlock() here ?

Reply:
👍 read lock should be fine. i will update in a new PR